e2e/test/iothub/method/MethodE2EPoolAmqpTests.cs

```diff
@@ -19,203 +19,45 @@ public class MethodE2EPoolAmqpTests : E2EMsTestBase
-                    PoolingOverAmqp.SingleConnection_PoolSize,
-                    PoolingOverAmqp.SingleConnection_DevicesCount,
-                    MethodE2ETests.SetDeviceReceiveMethodAsync)
-                .ConfigureAwait(false);
-        }
-
-        // TODO: #943 - Honor different pool sizes for different connection pool settings.
-        [Ignore]
-        [TestMethod]
-        public async Task Method_DeviceSak_DeviceReceivesMethodAndResponse_SingleConnection_AmqpWs()
-        {
-            await SendMethodAndRespondPoolOverAmqp(
-                    Client.TransportType.Amqp_WebSocket_Only,
-                    PoolingOverAmqp.SingleConnection_PoolSize,
-                    PoolingOverAmqp.SingleConnection_DevicesCount,
-                    MethodE2ETests.SetDeviceReceiveMethodAsync)
-                .ConfigureAwait(false);
-        }
-
-        // TODO: #943 - Honor different pool sizes for different connection pool settings.
-        [Ignore]
-        [TestMethod]
-        public async Task Method_IoTHubSak_DeviceReceivesMethodAndResponse_SingleConnection_Amqp()
-        {
-            await SendMethodAndRespondPoolOverAmqp(
-                    Client.TransportType.Amqp_Tcp_Only,
-                    PoolingOverAmqp.SingleConnection_PoolSize,
-                    PoolingOverAmqp.SingleConnection_DevicesCount,
-                    MethodE2ETests.SetDeviceReceiveMethodAsync,
-                    authScope: ConnectionStringAuthScope.IoTHub)
-                .ConfigureAwait(false);
-        }
-
-        // TODO: #943 - Honor different pool sizes for different connection pool settings.
-        [Ignore]
-        [TestMethod]
-        public async Task Method_IoTHubSak_DeviceReceivesMethodAndResponse_SingleConnection_AmqpWs()
-        {
-            await SendMethodAndRespondPoolOverAmqp(
-                    Client.TransportType.Amqp_WebSocket_Only,
-                    PoolingOverAmqp.SingleConnection_PoolSize,
-                    PoolingOverAmqp.SingleConnection_DevicesCount,
+        //[DataRow(Client.TransportType.Amqp_Tcp_Only, ConnectionStringAuthScope.Device, PoolingOverAmqp.SingleConnection_PoolSize, PoolingOverAmqp.SingleConnection_DevicesCount)]
+        //[DataRow(Client.TransportType.Amqp_WebSocket_Only, ConnectionStringAuthScope.Device, PoolingOverAmqp.SingleConnection_PoolSize, PoolingOverAmqp.SingleConnection_DevicesCount)]
+        //[DataRow(Client.TransportType.Amqp_Tcp_Only, ConnectionStringAuthScope.IoTHub, PoolingOverAmqp.SingleConnection_PoolSize, PoolingOverAmqp.SingleConnection_DevicesCount)]
+        //[DataRow(Client.TransportType.Amqp_WebSocket_Only, ConnectionStringAuthScope.IoTHub, PoolingOverAmqp.SingleConnection_PoolSize, PoolingOverAmqp.SingleConnection_DevicesCount)]
+        [DataRow(Client.TransportType.Amqp_Tcp_Only, ConnectionStringAuthScope.Device, PoolingOverAmqp.MultipleConnections_PoolSize, PoolingOverAmqp.MultipleConnections_DevicesCount)]
```

Comment:
Evaluate the time this takes running in serial to determine if you want to use this or not.

Reply:
If you give me a day, we should be able to get that data :)

Reply:
I'm fairly certain these tests take a long time to run in serial. I'm with Abhipsa on this one. I'm annoyed that MSTest isn't smart enough to parallelize these, though

Reply:
I don't have context on the time this takes but we should definitely consider thread starvation. We cannot run everything in parallel and we need to group some tests (may not be these). Also, adding so many tests is hard to really maintain/figure out what combinations are being tested. The other option is to combine these tests if possible. Again I have not looked at these tests to know if that is a good solution for this situation or nor. Plus currently we only have parallelism across classes and not methods 'Scope = ExecutionScope.ClassLevel'. Abhipsa tried to switch it to per test and it seemed much much slower. We should definitely investigate further if there is any advantage not to add [DataRow] if we really cannot switch to 'Scope = ExecutionScope.MethodLevel' or not.